internal/testing/conformance/consul-config.yaml

```diff
@@ -12,4 +12,4 @@ controller:
 apiGateway:
   enabled: true
   logLevel: info
-  image: "consul-api-gateway:test"
+  image: "hashicorppreview/consul-api-gateway:0.3.0-dev"
```

Comment:
Good to call out for ourselves that we're departing from building and testing whatever branch we run conformance tests on - we're now always testing the latest build from our `main` branch

Reply:
Is that what we want? At best that would be misleading when using the `pr/run-conformance` label on PRs...

Reply:
We'll probably wind up with a special workflow for PRs with the label that packages up the PR branch and runs with that, just re-using the building blocks that we already have for the nightly runs. I can put that together in < 10 minutes tomorrow.

Reply:
https://github.com/hashicorp/consul-api-gateway/pull/238